docs: adds info about pull queries on derived table (#7341)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/select-pull-query.md

If no bounds are placed on `WINDOWSTART` or `WINDOWEND`, rows are returned for all windows
in the windowed table.

> Pull queries can also be issued against derived tables created using [CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statements. 

Reviewer: ```suggestion
Pull queries can also be issued against derived tables created by using [CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statements. 

```

> Here is an example of that:

Reviewer: ```suggestion

```

```sql
CREATE TABLE GRADES (ID INT PRIMARY KEY, GRADE STRING, RANK INT) 
  WITH (kafka_topic = 'test_topic', value_format = 'JSON', partitions = 1);
```

> This creates a source table `GRADES`. Create a derived table

Reviewer: ```suggestion
Create a derived table, named 

```

> `TOP_TEN_RANKS` with a [CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statement:

Reviewer: ```suggestion
`TOP_TEN_RANKS`, my using a [CREATE TABLE AS SELECT](../../ksqldb-reference/create-table-as-select) statement:

```

 ```sql
CREATE TABLE TOP_TEN_RANKS 
  AS SELECT ID, RANK 
  FROM GRADES 
  WHERE RANK <= 10;
 ```

> We can fetch the current state of our materialized view 

Reviewer: ```suggestion
You can fetch the current state of your materialized view, 

```

> (the derived table `TOP_TEN_RANKS`) with pull queries:

Reviewer: ```suggestion
the `TOP_TEN_RANKS` derived table, by using a pull query:

```

```sql
SELECT * FROM TOP_TEN_RANKS;
```

> or, if you just want to just look up the student with `ID = 5` in the derived table:

Reviewer: ```suggestion
The following statement looks up only the student with `ID = 5` in the derived table:

```
